src/cdsetool/download.py

```diff
@@ -126,3 +143,60 @@ def _set_proxy(options, session):
     if proxies != {}:
         session.proxies.update(proxies)
     return session
+
+
+def validity_check(temp_path, product_info):
+    """Given it's temp_path and metadata info checks if the data downloaded is valid"""
+    size = os.path.getsize(temp_path)
+    content_length = (
+        product_info.get("properties").get("services").get("download").get("size")
+    )
+    if size > content_length:
+        # Dowloaded more than the size of the file.
+        return Validity.INVALID
+    if size == content_length:
+        checksum_comparison = _checksum_compare(temp_path, product_info)
+        if checksum_comparison is None:
+            # No data available for the checksum
+            return Validity.IGNORE
+        if checksum_comparison:
+            return Validity.VALID
+        # Checksum failed
+        return Validity.INVALID
+    # here it's a partial download logic to implement is TODO
+    return Validity.CONTINUE
+
+
+def _checksum_compare(temp_path, product_info):
+    """Compare a given MD5 checksum with one calculated from a file."""
+    checksum = None
```

Comment:
Remove this and the next line, `algo` and `checksum` are both initialized on line 175.